### ISSUE TITLE
`connectPool` never returns second time

### ISSUE THREAD

**Issue by GaikwadPratik** [rethinkdb/rethinkdb-ts#47]
@ronzeidman ,

This is more like a question than an issue. If I invoke `await r.connectPool(dbOptions);` multiple times(mistakenly), then only the first call returns, the second call or any subsequent calls to the above line never returns and kind of appear to be freezing.

Any idea, why?

**Comment by anli-xsigns** [comment]
Yes, I spent the whole day finding the issue I had with my code (I had two connectPool too). The problem is in lines https://github.com/rethinkdb/rethinkdb-ts/blob/12fa16308cb73239dc5947a7da1ae82fd5292d3e/src/query-builder/r.ts#L51-L54.
There an existing pool on the same connection will be drained if a new pool is created on the same connection. That can lead to silent errors in my opinion.

At least it is necessary to await the pool finishing the draining in line https://github.com/rethinkdb/rethinkdb-ts/blob/12fa16308cb73239dc5947a7da1ae82fd5292d3e/src/query-builder/r.ts#L53 before (r as any).pool is overwritten while draining in https://github.com/rethinkdb/rethinkdb-ts/blob/12fa16308cb73239dc5947a7da1ae82fd5292d3e/src/query-builder/r.ts#L59

Maybe it is better to throw an error that a pool has to be closed manually before creating a new one instead of silently creating a new pool to avoid clumsy errors.

**Comment by atassis** [comment]
@GaikwadPratik @anli-xsigns  I might resolve this by splitting the logic into separate models, so `r` won't be a singleton which rules them all. `r` will only be needed to generate a query, while all connection handling, query running, etc will be handled by different pure functions

**Comment by anli-xsigns** [comment]
At the moment I'm not using this anymore. Your idea sounds good.

**Comment by atassis** [comment]
@anli-xsigns you are not using rethinkdb or this exact library?

**Comment by anli-xsigns** [comment]
I'm not using rethinkdb anymore because work seems to be suspended again.

**Comment by atassis** [comment]
Can't say thats a bad decision=)
